src/it/resources/com/google/checkstyle/test/chapter4formatting/rule412nonemptyblocks/InputRightCurlySwitchCasesBlocks.java

```diff
@@ -72,15 +78,71 @@ public static void test3() {
     int k = 0;
     switch (k) {
       case 1:
-        { // violation ''{' at column 9 should be on the previous line.'
+        {
           int x = 1;
           break;
         }
-      case 2:
-        { // violation ''{' at column 9 should be on the previous line.'
-        int x = 2; } // violation ''}' at column 20 should be alone on a line.'
+      default:
+        {
+          int x = 2; } // false-negative
+    }
+  }
+
+  /** some javadoc. */
+  public static void test4() {
+    int mode = 0;
+    switch (mode) {
+      case 1: { // violation ''{' at column 15 should be on a new line.'
+          int x = 1; // violation '.* incorrect indentation level 10, expected level should be 8.'
+          break; // violation '.* incorrect indentation level 10, expected level should be 8.'
+          // 2 violations 3 lines below:
+          //  ''block rcurly' has incorrect indentation level 8, expected level should be 6.'
+          //  ''}' at column 9 should be alone on a line.'
+        } default:
+          { // violation '.* incorrect indentation level 10, expected level should be 8.'
+                  int x = 0; // violation '.* incorrect indentation .*, expected .* 8, 10.'
+          } // violation '.* incorrect indentation level 10, expected level should be 8.'
+    }
+  }
+
+  /** some javadoc. */
+  public static void test5() {
+    int k = 0;
+    switch (k) {
+      case 1: { // violation ''{' at column 15 should be on a new line.'
+        int x = 1; } // violation ''}' at column 20 should be alone on a line.'
```

Comment:
Similar case as above that you marked as false negative, but it is violation here, that is expected

Reply:
Above case was for `default` and this is for `case`, there's seem to be some bug in our implementation so we're not properly able to detect KR blocks for `default`

Reply:
Please respond with a link to the issue where the bug will be fixed

Reply:
https://github.com/checkstyle/checkstyle/issues/15664